test/test_flake8.py

```diff
@@ -11,13 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from ament_flake8.main import main
 import pytest
 
+from flake8.api.legacy import get_style_guide
+
 
 @pytest.mark.flake8
 @pytest.mark.linter
 def test_flake8():
-    rc = main(argv=[])
-    assert rc == 0, 'Found code style errors / warnings'
+    # Configuration options adapted from the ament_flake8
+    # https://github.com/ament/ament_lint/blob/master/ament_flake8/ament_flake8/configuration/ament_flake8.ini
+    style = get_style_guide(
+        ignore='C816,D100,D101,D102,D103,D104,D105,D106,D107,D203,D212,D404,I202',
```

Comment:
Not all of these are probably necessary. I would suggest to remove the ones not relevant for this package.

Reply:
In order to do that there needs to be a shared understanding of what error codes are relevant.

I can think of one heuristic which would be with all flake8 plugins from CI installed: Do any ignored checks trigger? The ones that do should stay and others can be removed.

Absent doing the above or some other heuristic test, I would need some criteria to decide which are applicable and audit each individual ignored rule.

Reply:
Currently CI passes for this packages. After the patch that should be the same - without modifying any code and without suppressing warning classes which are not even emitted.

Reply:
Okay so that's a preference for only ignoring warnings that are suppressed in the current codebase rather than suppressing warnings consistently across this package and packages that use ament_lint. :+1: